sklearnex/decomposition/pca.py

```diff
@@ -129,7 +129,7 @@ def _onedal_fit(self, X, queue=None):
             onedal_params = {
                 "n_components": self.n_components,
                 "is_deterministic": True,
-                "method": "cov",
+                "method": "cov" if self._fit_svd_solver == "covariance_eigh" else "svd",

```

Comment:
"svd" is slower than "cov" at the moment,  better to keep "cov" as default for now until "svd" is optimized. a separate investigation will be done to tune svd_solver for different data size

Reply:
When svd and covariance solvers are simultaneously available in sklearn API, always defaulting to covariance is direct inconsistency with sklearn and disinforms users about used implementation.

Reply:
Agree on the inconsistency but the svd has been implemented very recently. Would be nice to have a perf measurement and decide on the trade-off.

Reply:
BTW, "svd" is not an option for user input.

Reply:
Considering oneDAL's SVD perf., I decided to choose next approach for svd_solver dispatching:
- Leave previous behaviour for sklearn<1.5
- If `full` svd_solver is explicitly requested fallback to stock sklearn for versions>=1.5
- If `auto` svd_solver selection of stock sklearn dispatches to `full` or `covariance_eigh`, use oneDAL PCA-Covariance with warning for `full` solver

Reply:
The idea of explicit "full" dispatched to use SVD sounds good to me. Only thing is, oneDAL SVD solver should still outperform sklearn_svd_solver, unless you noticed otherwise. In that case, when explicit "full" is called, we can dispatch to onedal_svd.

Reply:
oneDAL SVD performance improvement is not consistent on all cases/configuration, so it is better to leave it out of patch until SVD perf. is fixed/improved.